include/SDL_egl.h

```diff
@@ -26,6 +26,8 @@
  */
 #if !defined(_MSC_VER) && !defined(__ANDROID__)
 
+#define MESA_EGL_NO_X11_HEADERS
```

Comment:
This causes redefinition warnings on Wayland, both because it's defined elsewhere in the source as well as the autoconf script!

This as well as src/video/wayland/SDL_waylandvideo.h needs to have `#ifndef` guards to avoid this. That said, this is a totally separate issue from what this patch covers, we just have an existing issue with these definitions that this uncovers. I'm not sure what the correct solution is, but this isn't strictly _wrong_, so I'm just pointing it out for future reference.

Reply:
This change was needed to build on a pure Wayland system with --disable-video-vulkan
Just note that when using Vulkan support the build was OK thanks to:

  #include "SDL_sysvideo.h"
    -> #include "SDL_vulkan_internal.h"
      -> #include "wayland/SDL_waylanddyn.h"
        -> #include "wayland-egl.h"
          -> #define WL_EGL_PLATFORM 1 done before eglplatform.h inclusion

Yes, the change was similar to the #define MESA_EGL_NO_X11_HEADERS done in SDL_waylandvideo.h before the inclusion of eglplatform.h

But I understand the conflict due to the -DMESA_EGL_NO_X11_HEADERS -DEGL_NO_X11 flags set by the configure script with --disable-video-x11: I updated the commit to fix it.

Nicolas Caramelli